repository_service_tuf/cli/admin/metadata.py

```diff
@@ -587,3 +605,117 @@ def update(
     else:
         # There are no changes made to the root metadata file.
         console.print("\nNo file will be generated as no changes were made\n")
+
+
+def _get_pending_signatures(
+    settings: Any, api_url: Optional[str]
+) -> Dict[str, Any]:
+    if settings.AUTH is False and api_url is None:
+        api_url = prompt.Prompt.ask("\n[cyan]API[/] URL address")
+        settings.SERVER = api_url
+
+    headers = get_headers(settings)
+    response = request_server(
+        settings.SERVER, URL.metadata_sign.value, Methods.get, headers=headers
+    )
+    if response.status_code != 200:
+        raise click.ClickException(
+            f"Failed to retrieve the sign data. Error: {response.text}"
+        )
+
+    response_data: Dict[str, Any] = response.json().get("data")
+    if response_data is None:
+        raise click.ClickException(response.text)
+
+    signing_roles: Dict[str, Any] = response_data.get("metadata", {})
+    if len(signing_roles) == 0:
+        raise click.ClickException("No metadata available for signing")
+
+    return signing_roles
+
+
+def _get_signing_key(role_info: MetadataInfo) -> RSTUFKey:
+    pending_keys: List = []
+    for key_id in role_info._new_md.signed.roles[Root.type].keyids:
+        if key_id not in role_info._new_md.signatures:
+            pending_keys.append(role_info._new_md.signed.keys[key_id])
+
+    sign_key_name = prompt.Prompt.ask(
+        "\nChoose a private key to load",
+        choices=[
+            signing_key.unrecognized_fields["name"]
```

Comment:
What if a user has forgotten to add a key or directly sends bootstrap information without using the CLI?
It's still possible to have a key without a name.
That's why I did that inside `load_key`:
https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/9a7643c872fafac74cd36fe99a76b85c007ccec5/repository_service_tuf/helpers/tuf.py#L420
You need to make sure each of the keys has some name.